tests/e2e/creation_test.go

```diff
@@ -349,4 +349,70 @@ func (tc *DSCTestCtx) ValidateHardwareProfileCR(t *testing.T) {
 		)),
 		WithCustomErrorMsg("Hardware profile was not recreated with default values"),
 	)
+
+	// verifed custom-serving hardwareprofile exists and api version is correct on v1.
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.HardwareProfile, types.NamespacedName{Name: "custom-serving", Namespace: tc.AppsNamespace}),
+		WithCondition(And(
+			jq.Match(`.spec.identifiers[0].identifier == "cpu"`),
+			jq.Match(`.spec.identifiers[0].displayName == "cpu"`),
+			jq.Match(`.spec.identifiers[0].resourceType == "CPU"`),
+			jq.Match(`.spec.identifiers[0].minCount == 1`),
+			jq.Match(`.spec.identifiers[0].defaultCount == 1`),
+			jq.Match(`.spec.identifiers[1].identifier == "memory"`),
+			jq.Match(`.spec.identifiers[1].displayName == "memory"`),
+			jq.Match(`.spec.identifiers[1].resourceType == "Memory"`),
+			jq.Match(`.spec.identifiers[1].minCount == "1Gi"`),
+			jq.Match(`.spec.identifiers[1].defaultCount == "1Gi"`),
+			jq.Match(`.metadata.annotations["opendatahub.io/managed"] == "false"`),
+			jq.Match(`.metadata.annotations["opendatahub.io/dashboard-feature-visibility"] == "model-serving"`),
+			jq.Match(`.apiVersion == "infrastructure.opendatahub.io/v1"`),
+		)),
+		WithCustomErrorMsg("Custom-serving hardwareprofile should have correct identifiers, managed=false, dashboard-feature-visibility=model-serving, and use v1 API version"),
+	)
+
+	// update custom-serving hardwareprofile to different value and check it is updated.
+	tc.EnsureResourceCreatedOrPatched(
```

Comment:
EnsureResourceCreatedOrPatched need to be updated here
the function has been renamed to EventuallyResourceCreatedOrUpdated